File: project-euler.java/src/main/java/euler/solutions/_10.java
package euler.solutions;

import euler.common.BaseSolution;

import static euler.common.NumberUtils.PrimesIterator;

public class _10 extends BaseSolution {

    @Override
    public long solve() {
        long sum = 2 + 3 + 5;

        PrimesIterator primes = new PrimesIterator();
        int x = primes.next();
        while (x < 2_000_000) {
            sum += x;
            x = primes.next();
        }

        return sum;
    }
}

File: project-euler.go/solutions/_10/10.go
package _10

import "project-euler/common/numbers"

func Solve() int64 {
	var sum int64 = 2 + 3 + 5

	primes := numbers.NewPrimesGenerator()
	x := primes.Next()
	for ; x < 2000000; x = primes.Next() {
		sum += int64(x)
	}

	return sum
}
